src/Analysis/Ast/Impl/Modules/PythonModule.cs

```diff
@@ -585,12 +658,25 @@ public Task<IDocumentAnalysis> GetAnalysisAsync(int waitTime = 200, Cancellation
         public virtual int LocationToIndex(SourceLocation location) => this.GetAst()?.LocationToIndex(location) ?? default;
         #endregion
 
+        #region IDependencyProvider
+        public virtual HashSet<AnalysisModuleKey> GetDependencies(PythonAst ast) {
+            var dbs = GetDbService();
+            if (dbs != null && dbs.TryRestoreDependencies(this, out var dp)) {
```

Comment:
So `TryRestoreDependencies` is used only here, and `GetDependencies` is used only in `PythonAnalyzerEntry`. It feels like this logic can be left inside `PythonAnalyzerEntry`.

Reply:
It was mostly to isolate dependency analysis from knowledge if particular module has AST or has other means of fetching dependencies. This can be extended to handle that `LoadFunctionDependencyModules` - module that has that kind of functions can supply additional dependencies in a normal way.

Reply:
Updated per discussion